lib/default.nix

```diff
@@ -37,6 +38,10 @@ lib.makeExtensible (
       byteCompileLuaDrv
       ;
 
+    inherit (self.dependencies)
+      mkDependencyModule
+      ;
```

Comment:
Do we need this at the top-level? `lib.nixvim.dependencies.mkDependencyModule` is probably fine as it's only used internally.

Alternatively, maybe we could move it to the plugin utils? (`lib.nixvim.plugins.utils`) Unless it's also used elsewhere?